### ISSUE TITLE
The `@stylistic/unit-case` rule reads two dimensions where a hexadecimal escape has swallowed the whitespace between them

### ISSUE THREAD

**Issue by firefoxic** [stylelint-stylistic/stylelint-stylistic#526]
### What minimal example or steps are needed to reproduce the bug?

A dimension whose hexadecimal escape swallows the whitespace behind it:

```css
a { width: 10px\9 2PX; }
```

### What minimal configuration is needed to reproduce the bug?

```json
{
  "plugins": [
    "@stylistic/stylelint-plugin"
  ],
  "rules": {
    "@stylistic/unit-case": "lower"
  }
}
```

### How did you run Stylelint?

```shell
stylelint "**/*.css" --config myconfig.json
```

### What did you expect to happen?

Nothing about `2PX`. CSS closes a hexadecimal escape with one whitespace character belonging to the escape rather than to the text, so `10px\9 2PX` is one dimension token whose unit is `px`, a tab, `2PX` — `@csstools/css-tokenizer` reads exactly that, and Sass, Less and `lightningcss@1.33.0` all print the line back exactly as it stands, none of them finding two values there.

### What actually happened?

```
1:20-22  Expected "PX" to be "px"
```

and `--fix` leaves `a { width: 10px\9 2px; }`. The two options disagree about which half of the one unit to change: under `upper` the same line is reported as `1:14-16  Expected "px" to be "PX"` and comes out of `--fix` as `a { width: 10PX\9 2PX; }`.

Under `@stylistic/scss/unit-case` and `@stylistic/less/unit-case` the same, the reading being plain CSS's.

### What is the cause?

`postcss-value-parser` breaks a value at whitespace and knows nothing of the escape that swallows it, so it hands the rule two words, `10px\9` and `2PX`, where the file spells one. `lib/rules/unit-case/index.ts` reads each word it is handed as a dimension of its own.

This is the whitespace half of the reading [#414](https://github.com/stylelint-stylistic/stylelint-stylistic/issues/414) settled for a single character: there an escape covering a star was read as an operator, here an escape covering a space is read as the end of a word.

### Do you have a proposal to fix the bug?

Weld back together the words a hexadecimal escape's own whitespace parted, before any of them is read as a dimension.

**Comment by firefoxic** [comment]
A second spelling of this reading, met on the branch that closes [#414](https://github.com/stylelint-stylistic/stylelint-stylistic/issues/414). The mechanism above is a hexadecimal escape swallowing the whitespace behind it, so that `postcss-value-parser` parts a word CSS spells as one. This one is the other way round: the parser hands over as one word what the grammar reads as several tokens, and a rule that reads the first of them says nothing about the rest.

A character that ends a unit does not always part a word. Every character that is no code point of an identifier ends one, and none of them parts the word that parser gives back: `10PX%2REM` is the dimension `10PX`, a delimiter and the dimension `2REM` to `@csstools/css-tokenizer`, and one word to the parser. The period and the plus do not even leave a delimiter — `10PX.2REM` and `10PX+2REM` are two dimension tokens standing next to each other, and `lightningcss@1.33.0` recases the units of both, printing `10px.2rem` and `10px+2rem`. A backslash in front of a line break goes the same way, opening no escape and parting no word.

Until that branch `unit-case` called the whole of such a word one unit and recased every letter of it, so the second dimension was reached by accident: `10PX%2REM` under `lower` came back `10px%2rem`. Since it the unit ends where its identifier does, which is what the grammar says, and the second dimension is reached by nothing: the same value comes back `10px%2REM`. Measured as what `--fix` leaves behind — a dimension token whose raw unit is still in the case the option does not ask for, where the base's fix corrected it — 585 rows of `scripts/sweeps/escaped-unit-end.ts` stand on this shape, over the eight characters that sweep spells.

So both spellings are the same thing — the words that parser hands over are not the tokens the file spells — and a fix for this issue wants to answer for both.
